### PR TITLE
feat: implement new generic error page

### LINKED ISSUE #2958: unified error page
### Is your feature request related to a problem?

Currently we have a bit ugly error page for missing collections

### Describe the solution you would like

I would like to see a universal page for errors like `/404` where you add can add the query params like `?type=collection&reason=error.notfound&id=dasdas` so we can reuse it also for missing nfts or other pages :)

### Describe alternatives you have considered

It can be built on top of the `Error.vue` component :)

### Screeenshots

`errorcollection`

Screenshot 2022-05-06 at 11 54 53

`current 404`

Screenshot 2022-05-06 at 12 00 22

`Error.vue`

Screenshot 2022-05-06 at 12 00 45

### PR DESCRIPTION
**Pull request by kkukelka**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check

### PR type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

Screenshot 2022-05-06 at 20 04 49
Screenshot 2022-05-06 at 20 05 15
eck before the contribution.

### What's new?

- [x] PR closes #2958
- [x] get rid of old error handling mess and implement correct implementation for nuxt;
this will get even more beneficial once we utilize asyncData for fetching

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://kodadot.xyz/transfer/?target=EqdyzrzVmeHwMdMwvPeCMnNdbuQDbD3YrjY93xq9Ln3jUGW)

### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

**Comment by vikiival** [comment]
Really nice :)

few thingies: 

Can we have a cat for each http err ? (e.g based on the status code ?)

also it is possible to have this? 

> query params like ?type=collection&reason=error.notfound&id=dasdas

It was written in the issue. 

The point is when user got a error screen it can tell which collection / nft caused that err :)

**Review by vikiival — COMMENTED** [review]
.

**Comment by kkukelka** [comment]
> Really nice :)
> 
> few thingies:
> 
> Can we have a cat for each http err ? (e.g based on the status code ?)
> 
> also it is possible to have this?
> 
> > query params like ?type=collection&reason=error.notfound&id=dasdas
> 
> It was written in the issue.
> 
> The point is when user got a error screen it can tell which collection / nft caused that err :)

yes, NuxtError interface allows to propagate the path where the error occured.

Screenshot 2022-05-08 at 19 53 28

just implemented it for collections

so wherever we want to explicitly track the URL that caused failure we just need to implement 
```javascript
  return this.$nuxt.error({
        statusCode: 404,
        message: 'Oops! Collection Not Found',
        path: this.$route.path,
      })
```

File: layouts/error.vue [hunk 1]
```diff
@@ -1,22 +1,52 @@
...
-    <p class="subtitle">Indexer is not working properly.</p>
-    <img src="https://http.cat/500" alt="Internal error cat" />
+  <div id="Error" class="box container has-text-centered">
+    <h1 class="title">{{ error.statusCode }} - {{ headline }}</h1>
+    <p class="subtitle">{{ error.message || 'Something went wrong' }}</p>
+    <p v-if="error.path" class="subtitle">Path: {{ error.path }}</p>
+    <!-- <p class="subtitle">Indexer is not working properly.</p> -->
+    <img
+      v-if="error.statusCode === 500"
+      src="https://http.cat/500"
```
Comment (vikiival) [inline layouts/error.vue:9]:
```suggestion
      src="`https://http.cat/${error.statusCode}`"
```

**Review by vikiival — CHANGES_REQUESTED** [review]

**Comment by kkukelka** [comment]
@vikiival very neat, didnt know it had a picture for every status code 😎

File: layouts/error.vue [hunk 2]
```diff
@@ -1,22 +1,51 @@
...
-    <p class="title">Indexer Error</p>
-    <p class="subtitle">Indexer is not working properly.</p>
-    <img src="https://http.cat/500" alt="Internal error cat" />
+  <div id="Error" class="box container has-text-centered">
+    <h1 class="title">{{ error.statusCode }} - {{ headline }}</h1>
+    <p class="subtitle">{{ error.message || 'Something went wrong' }}</p>
+    <p v-if="error.path" class="subtitle">Path: {{ error.path }}</p>
+    <!-- <p class="subtitle">Indexer is not working properly.</p> -->
+    <img
+      src="`https://http.cat/${error.statusCode}`"
```
Comment (vikiival) [inline layouts/error.vue:8]:
```suggestion
      :src="`https://http.cat/${error.statusCode}`"
```

**Review by vikiival — APPROVED** [review]
Looks oki

**Comment by vikiival** [comment]
pay 75 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $75 @ 100.41 USD/KSM ~ 0.747 $KSM
🧗 [EqdyzrzVmeHwMdMwvPeCMnNdbuQDbD3YrjY93xq9Ln3jUGW](https://kusama.subscan.io/account/EqdyzrzVmeHwMdMwvPeCMnNdbuQDbD3YrjY93xq9Ln3jUGW)
🔗 [0x6a7a26810a8682b6fc7cf00c2b01266c1111c613da88e2673f700815fee19322](https://kusama.subscan.io/extrinsic/0x6a7a26810a8682b6fc7cf00c2b01266c1111c613da88e2673f700815fee19322)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**